test/other_tests.jl

```diff
@@ -63,13 +63,13 @@ end
     @test_throws MethodError zeta(big"1.0",2.0)
     @test_throws MethodError zeta(big"1",2.0)
 
-    @test typeof(polygamma(3, 0x2)) == Float64
-    @test typeof(polygamma(big"3", 2f0)) == Float32
-    @test typeof(zeta(1, 2.0)) == Float64
-    @test typeof(zeta(1, 2f0)) == Float64 # BitIntegers result in Float64 returns
-    @test typeof(zeta(2f0, complex(2f0,0f0))) == Complex{Float32}
-    @test typeof(zeta(complex(1,1), 2f0)) == Complex{Float64}
-    @test typeof(zeta(complex(1), 2.0)) == Complex{Float64}
+    @test polygamma(3, 0x2) isa Float64
+    @test polygamma(big"3", 2f0) isa Float32
+    @test zeta(1, 2.0) isa Float64
+    @test zeta(1, 2f0) isa Float32
```

Comment:
With this PR the result is a `Float32` whereas currently it is a `Float64`. I think the behaviour in this PR is expected and less surprising - usually, when one works with `Float32`/`Float16`, eg on the GPU, one does not want a result of type `Float64` but of type `Float32`/`Float16`. Therefore we explicitly convert the result of functions with one argument to `Float32`/`Float16`.

If one intends to "fix" the inconsistency with `BigInt` that these tests check for (I'm not sure if it is useful or desired), IMO one should rather handle `BigInt` arguments and fix the return types in this case instead of changing the arguably correct result types with regular integers.

Reply:
Maybe I miss something but I also don't think https://github.com/JuliaLang/julia/issues/17474 applies here. The concern there was that

> if `f(::Int64)::Float64` then `f(::BigInt)::BigFloat`

should be ensured. However, here `Base.Fix2(zeta, ::Float32)(::Int)::Float32` (a function with one integer argument obtained by fixing the second argument of `zeta` with a `Float32`) would not be covered by this implication since the version with regular integers does not return a `Float64`. If the intention is more generally that

> if `f(::Int)::Union{Float16,Float32,Float64}` then `f(::BigInt)::BigFloat`

then it is also satisfied for the change in this PR, i.e., if `zeta(::Int, ::Float32)::Float32`.